addons/mail/models/mail_link_preview.py

```diff
@@ -142,3 +147,8 @@ def _gc_mail_link_preview(self):
             ('message_id', '=', False),
             ('create_date', '<', fields.Datetime.now() - timedelta(days=lifetime)),
         ], order='create_date ASC', limit=1000).unlink()
+
+    @api.model
+    def _init_ignore_list(self):
+        base_url = urlparse(self.get_base_url())
+        self.env["ir.config_parameter"].set_param("mail.mail_link_preview_ignore_list", f".*{base_url.hostname}:{base_url.port}/(web|odoo)")
```

Comment:
We also need to handle cases where `urlparse` fails on base url. It's also "user input" after all.

We wouldn't want an incorrect one from preventing to init the module.

Reply:
It should fail silently now.